include/openthread/thread.h

```diff
@@ -174,7 +174,12 @@ typedef struct otMleCounters
     uint16_t mAttachAttempts;                ///< Number of attach attempts while device was detached.
     uint16_t mPartitionIdChanges;            ///< Number of changes to partition ID.
     uint16_t mBetterPartitionAttachAttempts; ///< Number of attempts to attach to a better partition.
-
+    uint64_t mDisabledTime; ///< Number of milliseconds device has been in OT_DEVICE_ROLE_DISABLED role.
+    uint64_t mDetachedTime; ///< Number of milliseconds device has been in OT_DEVICE_ROLE_DETACHED role.
+    uint64_t mChildTime;    ///< Number of milliseconds device has been in OT_DEVICE_ROLE_CHILD role.
+    uint64_t mRouterTime;   ///< Number of milliseconds device has been in OT_DEVICE_ROLE_ROUTER role.
+    uint64_t mLeaderTime;   ///< Number of milliseconds device has been in OT_DEVICE_ROLE_LEADER role.
+    uint64_t mTrackedTime;  ///< Total number of milliseconds tracked by previous counters.
```

Comment:
No need to add `#if` check since we cannot use `#if OPENTHREAD_CONFIG_{}` in public OT headers. 
We can document that certain APIs requires certain configs.

The background is that the header/file where the OT configs are defined can not be always provided as a public OT header (so to make it available to other libraries (sw modules) that may want to include the OT public headers).

Reply:
Makes sense, thanks. I added the note here and in CLI readme.